src/routers/api.ts

```diff
@@ -204,18 +228,43 @@ router.post('/transfers', async (req, res, next) => {
     if (!utils.regexp.FILE_KEY.test(req.body.path) || utils.regexp.CONSECUTIVE_DOTS.test(req.body.path)) {
       throw new RequestError('Invalid path', 400);
     }
-    if (req.body.recipient === undefined) {
+    await blobsHandler.retreiveMetadata(path.join(utils.constants.DATA_DIRECTORY, utils.constants.BLOBS_SUBDIRECTORY, req.body.path));
```

Comment:
I think this is a bug - you are prepending the data directory prefix here, but then the `retreiveMetadata` function prepends the same prefix again.

Side note - `retrieve` is misspelled here.

Reply:
@awrichar good catch! I fixed that issue and also addressed the typo, plus I updated the headers. All the changes are in https://github.com/hyperledger/firefly-dataexchange-https/pull/62